src/java.base/share/classes/java/util/TreeMap.java

```diff
@@ -382,15 +382,13 @@ final Entry<K,V> getEntry(Object key) {
         @SuppressWarnings("unchecked")
             Comparable<? super K> k = (Comparable<? super K>) key;
         Entry<K,V> p = root;
-        while (p != null) {
-            int cmp = k.compareTo(p.key);
-            if (cmp < 0)
-                p = p.left;
-            else if (cmp > 0)
-                p = p.right;
-            else
-                return p;
-        }
+        int cmp = k.compareTo(p.key);
+        if (cmp < 0)
+            p = p.left;
+        else if (cmp > 0)
+            p = p.right;
+        else
+            return p;
```

Comment:
Removing `while (p != null) {` is definitely wrong here as there is reassignment `p = p.left;` and `p = p.right;` controlling the loop behaviour.